From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: EJ Hsu <ejh@nvidia.com>
Date: Thu, 4 Jul 2024 12:52:30 +0000
Subject: [PATCH 153/245] NVIDIA: SAUCE: usb: host: xhci: Add Tegra264 virtual
 function configs

Add SOC data for Tegra264 VFs. XHCI driver use this info to know which
VF it's running on.

Bug 4295138

Change-Id: I42fa1c60e3fbc627fcaf468dc7b390dcfd37e3f7
Signed-off-by: EJ Hsu <ejh@nvidia.com>
Signed-off-by: Wayne Chang <waynec@nvidia.com>
Reviewed-on: https://git-master.nvidia.com/r/c/3rdparty/canonical/linux-noble/+/3436313
Reviewed-by: svcacv <svcacv@nvidia.com>
Tested-by: Henry Lin <henryl@nvidia.com>
Reviewed-by: Brad Griffis <bgriffis@nvidia.com>
GVS: buildbot_gerritrpt <buildbot_gerritrpt@nvidia.com>
Reviewed-by: Henry Lin <henryl@nvidia.com>
(cherry picked from commit 4b1ba9b521871153896b8961e2767f88d39d2299)

diff --git a/drivers/usb/host/xhci-tegra.c b/drivers/usb/host/xhci-tegra.c
index a5ff601e2b400..ea077047bd75b 100644
--- a/drivers/usb/host/xhci-tegra.c
+++ b/drivers/usb/host/xhci-tegra.c
@@ -2,7 +2,7 @@
 /*
  * NVIDIA Tegra xHCI host controller driver
  *
- * SPDX-FileCopyrightText: Copyright (c) 2014-2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-FileCopyrightText: Copyright (c) 2014-2025, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
  * Copyright (C) 2014 Google, Inc.
  */
 
@@ -3482,7 +3482,7 @@ static const struct tegra_xusb_soc tegra234_soc = {
 	.has_pg_support = true,
 };
 
-static const struct tegra_xusb_soc tegra234_vf1_soc = {
+static const struct tegra_xusb_soc tegra_vf1_soc = {
 	.vf_id = 1,
 	.is_xhci_vf = true,
 	.phy_types = tegra194_phy_types,
@@ -3494,7 +3494,7 @@ static const struct tegra_xusb_soc tegra234_vf1_soc = {
 	.lpm_support = true,
 };
 
-static const struct tegra_xusb_soc tegra234_vf2_soc = {
+static const struct tegra_xusb_soc tegra_vf2_soc = {
 	.vf_id = 2,
 	.is_xhci_vf = true,
 	.phy_types = tegra194_phy_types,
@@ -3506,7 +3506,7 @@ static const struct tegra_xusb_soc tegra234_vf2_soc = {
 	.lpm_support = true,
 };
 
-static const struct tegra_xusb_soc tegra234_vf3_soc = {
+static const struct tegra_xusb_soc tegra_vf3_soc = {
 	.vf_id = 3,
 	.is_xhci_vf = true,
 	.phy_types = tegra194_phy_types,
@@ -3518,7 +3518,7 @@ static const struct tegra_xusb_soc tegra234_vf3_soc = {
 	.lpm_support = true,
 };
 
-static const struct tegra_xusb_soc tegra234_vf4_soc = {
+static const struct tegra_xusb_soc tegra_vf4_soc = {
 	.vf_id = 4,
 	.is_xhci_vf = true,
 	.phy_types = tegra194_phy_types,
@@ -3530,6 +3530,42 @@ static const struct tegra_xusb_soc tegra234_vf4_soc = {
 	.lpm_support = true,
 };
 
+static const struct tegra_xusb_soc tegra_vf5_soc = {
+	.vf_id = 5,
+	.is_xhci_vf = true,
+	.phy_types = tegra194_phy_types,
+	.num_types = ARRAY_SIZE(tegra194_phy_types),
+	.ports = {
+		.usb3 = { .offset = 0, .count = 4, },
+		.usb2 = { .offset = 4, .count = 4, },
+	},
+	.lpm_support = true,
+};
+
+static const struct tegra_xusb_soc tegra_vf6_soc = {
+	.vf_id = 6,
+	.is_xhci_vf = true,
+	.phy_types = tegra194_phy_types,
+	.num_types = ARRAY_SIZE(tegra194_phy_types),
+	.ports = {
+		.usb3 = { .offset = 0, .count = 4, },
+		.usb2 = { .offset = 4, .count = 4, },
+	},
+	.lpm_support = true,
+};
+
+static const struct tegra_xusb_soc tegra_vf7_soc = {
+	.vf_id = 7,
+	.is_xhci_vf = true,
+	.phy_types = tegra194_phy_types,
+	.num_types = ARRAY_SIZE(tegra194_phy_types),
+	.ports = {
+		.usb3 = { .offset = 0, .count = 4, },
+		.usb2 = { .offset = 4, .count = 4, },
+	},
+	.lpm_support = true,
+};
+
 static const struct tegra_xusb_soc tegra264_soc = {
 	.supply_names = tegra194_supply_names,
 	.num_supplies = ARRAY_SIZE(tegra194_supply_names),
@@ -3564,11 +3600,18 @@ static const struct of_device_id tegra_xusb_of_match[] = {
 	{ .compatible = "nvidia,tegra186-xusb", .data = &tegra186_soc },
 	{ .compatible = "nvidia,tegra194-xusb", .data = &tegra194_soc },
 	{ .compatible = "nvidia,tegra234-xusb", .data = &tegra234_soc },
-	{ .compatible = "nvidia,tegra234-xusb-vf1", .data = &tegra234_vf1_soc },
-	{ .compatible = "nvidia,tegra234-xusb-vf2", .data = &tegra234_vf2_soc },
-	{ .compatible = "nvidia,tegra234-xusb-vf3", .data = &tegra234_vf3_soc },
-	{ .compatible = "nvidia,tegra234-xusb-vf4", .data = &tegra234_vf4_soc },
+	{ .compatible = "nvidia,tegra234-xusb-vf1", .data = &tegra_vf1_soc },
+	{ .compatible = "nvidia,tegra234-xusb-vf2", .data = &tegra_vf2_soc },
+	{ .compatible = "nvidia,tegra234-xusb-vf3", .data = &tegra_vf3_soc },
+	{ .compatible = "nvidia,tegra234-xusb-vf4", .data = &tegra_vf4_soc },
 	{ .compatible = "nvidia,tegra264-xusb", .data = &tegra264_soc },
+	{ .compatible = "nvidia,tegra264-xusb-vf1", .data = &tegra_vf1_soc },
+	{ .compatible = "nvidia,tegra264-xusb-vf2", .data = &tegra_vf2_soc },
+	{ .compatible = "nvidia,tegra264-xusb-vf3", .data = &tegra_vf3_soc },
+	{ .compatible = "nvidia,tegra264-xusb-vf4", .data = &tegra_vf4_soc },
+	{ .compatible = "nvidia,tegra264-xusb-vf5", .data = &tegra_vf5_soc },
+	{ .compatible = "nvidia,tegra264-xusb-vf6", .data = &tegra_vf6_soc },
+	{ .compatible = "nvidia,tegra264-xusb-vf7", .data = &tegra_vf7_soc },
 	{ },
 };
 MODULE_DEVICE_TABLE(of, tegra_xusb_of_match);
-- 
2.43.0

